Implemented fix for movie and album shared element transitions

poisdeux opened the pull request:
When shared element view is not visible on returning, the return shared
element transition starts off screen. This looks odd and we now simply
do not perform the shared element transition when the shared element is
not visible.

SyncedSynapse commented:
Thanks for this. I was thinking that it would have to be done soon.

This doesn't go in seamlessly because it includes changes to UIUtils.java, which are already present in the previous PR. Can you remove the changes to UIUtils.java in this one.

By the way, i know that in a previous discussion about code indention we agreed not to change Android Studio's defaults, but the the vast majority of the code is using the set of rules that i used, and now a lot of the changes are due to these different indentation settings. I would ask you to change your Android Studio indentation preferences, to avoid these unnecessary changes.
In Settings->Editor->Code Style->Java, check all "Align when multiline" (i will do to).

poisdeux commented:
Strange, I had no problem merging with the updated master. Should merge fine now.

MartijnKaijser commented:
please squash both commits into one to get rid of the "	Merge branch 'master' into transitionfix"

poisdeux commented:
I think that will really mess up the commit history. So, unless you can provide me with some details on how to squash the master merge commit together with my original commit, I suggest we keep it this way.
However, what I can do is change the commit message of the last commit. Do you have any suggestions?

MartijnKaijser commented:
We do it all the time
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

poisdeux commented:
Sure, I do too. The problem here is that after I merged master commit b48d6398be55f75e4b80a508428b3513e27770ea has two parents: fce521a9bbcd081081f25a7e36c512a92c3ffcda and 6485da94f2f65d48bac5c2893a22a3a8f1b18c83

Run 
git log --decorate --graph --all 
to see what I mean.
Their common ancestor is 9b4c4d6315b472efa84b6b2be4b6ad87dd9b49b6 and if I rebase it will include commit 5ccd76b349e734ebc1ca8e312b129519e948ee80 (from @SyncedSynapse ) as well. I have no clue on how to squash my two commits without also squashing 5ccd76b349e734ebc1ca8e312b129519e948ee80

MartijnKaijser commented:
guess rebase went wrong.
rebranch, apply your initial patch, force push to this branch

poisdeux commented:
?!?! There was no rebase, it was a merge. Nothing went wrong. This is how git works.

What I normally would have done is rebase instead of merging with master.
However, as I already published this branch in a PR, I didn't want to do a rebase as I was not sure if anyone else already pulled in this branch.

I also do not understand what the problem is as the commit history contains similar commits:

ce68ccbeb62a6e66d47bf9a91eb5979852324b95
fa77d460b63b6497c4eaa9ec4a70787acd4e92d7
69b1a9fcd4d1456e4409265a4e6fceb3b03121b4
85155ca040a667f3d7975ebe3d32aae13991f564

MartijnKaijser commented:
This is not how git works. You should ALWAYS rebase and never merge and after rebase force push this branch again. That's how git works. If some one pulled this branch that's his problem. 
That it happened in the past doesn't mean it should happen again.

poisdeux commented:
Seriously?!?

http://git-scm.com/book/en/v2/Git-Branching-Rebasing#The-Perils-of-Rebasing

I can see you are involved with Kodi, not sure what your involvement with
Kore is. Are you @syncedsynapse as well? If not, I'll wait on what
@syncedsynapse prefers before I rebranch and close this PR and create a new
PR.


On Fri, Dec 11, 2015 at 9:58 AM, Martijn Kaijser <notifications@github.com>
wrote:

> This is not how git works. You should ALWAYS rebase and never merge and
> after rebase force push this branch again. That's how git works. If some
> one pulled this branch that's his problem.
> That it happened in the past doesn't mean it should happen again.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/Kore/pull/146#issuecomment-163879187>.
>

MartijnKaijser commented:
I am the project manager for Kodi and admin for the XBMC github project in which Kore also is located. The PR as is would never, ever be merged into Kodi core code like this. You always need to rebase you commits on top of master/origin (or any upstream branch) and force push your changes to the PR.

Als you don't need to close the PR as you can simply force push any local branch you have to the branch of this PR.

SyncedSynapse commented:
Kore should be using the same standards as Kodi. It's currently not following those because of my ineptitude with git, but it should nevertheless.

@poisdeux can you please rebase it as @MartijnKaijser suggests.

poisdeux commented:
Ok, I will.

Would be nice if you could document these policies. Can't find anything
about rebase being required for Kodi development.

http://kodi.wiki/view/HOW-TO_submit_a_patch

Am I missing something?


On Fri, Dec 11, 2015 at 11:22 AM, Synced Synapse <notifications@github.com>
wrote:

> Kore should be using the same standards as Kodi. It's currently not
> following those because of my ineptitude with git, but it should
> nevertheless.
>
> @poisdeux <https://github.com/poisdeux> can you please rebase it as
> @MartijnKaijser <https://github.com/MartijnKaijser> suggests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/Kore/pull/146#issuecomment-163902729>.
>

poisdeux commented:
Ok, I rewrote the branch history into a single commit.

@SyncedSynapse About the Android settings, I changed them as requested. How should we proceed? Update the files as we go along, or do one PR that changes all files to the 'Align when multiline' setting?

SyncedSynapse commented:
Update as we go along. If you happen to touch a piece of code indent it as part of the process.
